include/coap2/net.h

```diff
@@ -844,4 +724,265 @@ unsigned int coap_calc_timeout(coap_session_t *session, unsigned char r);
+ *                   before returning. If COAP_RUN_BLOCK, the call will block
+ *                   until at least one new packet is received. If
+ *                   COAP_RUN_NONBLOCK, the function will return immediately
+ *                   following without waiting for any new input not already
+ *                   available.
+ * @param nfds      The maximum FD set in readfds, writefds or exceptfds
+ *                  plus one,
+ * @param readfds   Read FDs to additionally check for in internal select()
+ *                  or NULL if not required.
+ * @param writefds  Write FDs to additionally check for in internal select()
+ *                  or NULL if not required.
+ * @param exceptfds Except FDs to additionally check for in internal select()
+ *                  or NULL if not required.
+ *
+ *
+ * @return Number of milliseconds spent in coap_io_process_with_fds, or @c -1
+ *         if there was an error.  If defined, readfds, writefds, exceptfds
+ *         are updated as returned by the internal select() call.
+ */
+int coap_io_process_with_fds(coap_context_t *ctx, unsigned int timeout_ms,
+                        int nfds, fd_set *readfds, fd_set *writefds,
+                        fd_set *exceptfds);
+
+/**
+* Iterates through all the coap_socket_t structures embedded in endpoints or
+* sessions associated with the @p ctx to determine which are wanting any
+* read, write, accept or connect I/O (COAP_SOCKET_WANT_xxx is set). If set,
+* the coap_socket_t is added to the @p sockets.
+*
+* Any now timed out delayed packet is transmitted, along with any packets
+* associated with requested observable response.
+*
+* In addition, it returns when the next expected I/O is expected to take place
+* (e.g. a packet retransmit).
+*
+* Prior to calling coap_io_do_io(), the @p sockets must be tested to see
+* if any of the COAP_SOCKET_WANT_xxx have the appropriate information and if
+* so, COAP_SOCKET_CAN_xxx is set. This typically will be done after using a
+* select() call.
+*
+* Note: If epoll support is compiled into libcoap, coap_io_prepare_epoll() must
+* be used instead of coap_io_prepare_io().
+*
+* Internal function.
+*
+* @param ctx The CoAP context
+* @param sockets Array of socket descriptors, filled on output
+* @param max_sockets Size of socket array.
+* @param num_sockets Pointer to the number of valid entries in the socket
+*                    arrays on output.
+* @param now Current time.
+*
+* @return timeout Maxmimum number of milliseconds that can be used by a
+*                 select() to wait for network events or 0 if wait should be
+*                 forever.
+*/
+unsigned int
+coap_io_prepare_io(coap_context_t *ctx,
+  coap_socket_t *sockets[],
+  unsigned int max_sockets,
+  unsigned int *num_sockets,
+  coap_tick_t now
+);
+
+/**
+* @deprecated Use coap_io_prepare_io() instead.
+*
+* Iterates through all the coap_socket_t structures embedded in endpoints or
```

Comment:
I suggest also to either keep the old description or just refer to `coap_io_prepare_io()`.

Reply:
Updated to jut refer to coap_io_prepare_io() instead.